Source/HTTP/Curl/CurlMulti.cpp

```diff
@@ -94,16 +94,22 @@ HRESULT CALLBACK CurlMulti::CleanupAsyncProvider(XAsyncOp op, const XAsyncProvid
         assert(multi->m_cleanupAsyncBlock == nullptr);
         multi->m_cleanupAsyncBlock = data->async;
 
-        if (multi->m_taskQueueCallbacksPending == 0)
+        // If there are no pending task queue callbacks (and thus no pending HTTP requets), schedule cleanup immediately.
+        // If this condition is true, we're also guaranteed to be the final remaining reference to the CurlMulti.
+        bool cleanupImmediately = multi->m_taskQueueCallbacksPending == 0;
+        XTaskQueueHandle queue = multi->m_queue;
+
+        // Release the lock before going any further because the multi may be destroyed before Begin completes
+        lock.unlock();
```

Comment:
it might help to clarify how/why within this comment

```
// Release the lock before scheduling any further async work. That async work may drive the next XAsyncOp::DoWork iteration of this loop before this XAsyncOp::Begin iteration completes. Because DoWork cleans up the "multi" object it becomes no longer safe to reference that multi object after the async work is scheduled.
```

Reply:
Reworded slightly but I made the comment more detailed.